httpcache.go

```diff
@@ -106,6 +257,12 @@ func NewTransport(c Cache) *Transport {
 	return &Transport{Cache: c, MarkCachedResponses: true}
 }
 
+// SetLogging has the same parameters as the log.New function and replaces the
+// default logger that discards messages
+func (t *Transport) SetLogging(out io.Writer, prefix string, flags int) {
```

Comment:
Why not just have this take a `*log.Logger` for simplicity (instead of duplicating the `log.New` params)? Also, document that this is not thread-safe.